app/src/main/java/nextstep/shoppingcart/designsystem/component/ProductImage.kt

```diff
@@ -0,0 +1,51 @@
+package nextstep.shoppingcart.designsystem.component
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import coil3.compose.AsyncImagePainter
+import coil3.compose.rememberAsyncImagePainter
+import nextstep.shoppingcart.model.Product
+
+
+@Composable
+fun ProductImage(
+    item: Product,
+    modifier: Modifier = Modifier,
+    painter: AsyncImagePainter = rememberAsyncImagePainter(item.imageUrl)
+) {
+    val state by painter.state.collectAsState()
+
+    Box(
+        modifier = modifier
+            .fillMaxWidth()
+            .aspectRatio(1f), contentAlignment = Alignment.Center
+    ) {
+        when (state) {
+            is AsyncImagePainter.State.Loading, AsyncImagePainter.State.Empty -> {
+                CircularProgressIndicator(modifier = Modifier.testTag("product_image_loading"))
```

Comment:
```suggestion
                CircularProgressIndicator(modifier = Modifier.testTag("product_image_loading"))
                
                CircularProgressIndicator(
                    modifier = Modifier.semantics {
                        contentDescription = "product_image_loading"
                    }
                )
```

이런식으로 `testTag` 대안으로 사용하는 방식도 존재를 하는데요. 아무래도 View 시스템과 다르게 id가 필요없어졌다보니 `testTag` 를 사용하는데 살짝 어색함이 가끔 있더라고요! 

그래서 저는 `contentDescription` 을 활용하기도 하는데요. 그럼 무조건 이 방식으로 가는게 좋다 생각했는데, 막상 접근성을 대응하다보니 너무 많은 contentDescription 때문에 오히려 testTag로 변경도 하는 일이 생기더라고요. 그렇기에 덕성님만의 생각으로 어떤 것을 선택하실지 앞으로 고민해보시면 좋을거 같습니다.

> 오히려 접근성을 챙겨야 하는 곳이라면 테스트 할 겸 넣는 좋은 상황이 되기도 합니다 👍

---

추가로 관련 블로그글 하나 추천드립니다!

<img width="741" alt="image" src="https://github.com/user-attachments/assets/96634cfb-7d97-4bff-8d51-a0817e47636d" />

> https://callmeryan.medium.com/efficient-testing-or-code-pollution-the-role-of-testtag-in-jetpack-compose-testing-4892ee73e966

Reply:
엇.. 제가 원하던걸 알려주셨네요..

저도 testTag 를 별로 좋아하지 않는데 contentDescription 을 require 한 곳에서는 contentDescription 로 테스트를 하면되는데

저렇게 아예 semantics 를 이용해 contentDescription 를 작성하는것도 있군요..

확장함수같은걸로 만들어 놓고 쓰면 좋을것 같은 느낌이 드네요!

감사합니다!